integration-tests/tests/src/utils/ExtendedAppConfigBuilder.ts

```diff
@@ -0,0 +1,231 @@
+      type: "custom-function",
+      config: {
+        authFunctionName: "customAuthentication",
+      },
+      disabled: false,
+    }).function({
+      name: "customAuthentication",
+      private: true,
+      run_as_system: true,
+      source,
+    });
+  }
+
+  customTokenAuth({
+    privateKey,
+    metadataFields = [],
+  }: {
+    privateKey: string;
+    metadataFields: Array<CustomTokenAuthMetadataField>;
+  }) {
+    return this.secret("jwt-secret", privateKey).authProvider({
+      name: "custom-token",
+      type: "custom-token",
+      config: {
+        audience: "",
+        signingAlgorithm: "HS256",
+        useJWKURI: false,
+      },
+      secret_config: {
+        signingKeys: ["jwt-secret"],
+      },
+      disabled: false,
+      metadata_fields: metadataFields,
+    });
+  }
+
+  mongodbService(syncConfig: MongoServiceSyncConfig = { kind: "disabled" }, developmentMode = true) {
+    this.developmentMode(developmentMode);
+    return this.service(
+      {
+        name: "mongodb",
+        type: mongodbServiceType,
+        config: buildMongoDBConfig(syncConfig),
+        secret_config: {
+          uri: "mongodb_uri",
+        },
+      },
+      [
+        {
+          roles: [
+            {
+              name: "defaultRole",
+              apply_when: {},
+              document_filters: {
+                read: true,
+                write: true,
+              },
+              write: true,
+              read: true,
+              insert: true,
+              delete: true,
+            },
+          ],
+        },
+      ],
+    ).secret("mongodb_uri", "mongodb://localhost:26000");
+  }
+
+  partitionBasedSync(config: Partial<PartitionConfig> = {}, developmentMode = true) {
+    return this.mongodbService(
+      {
+        kind: "partition-based",
+        partition: {
+          key: "partition",
+          type: "string",
+          required: false,
+          permissions: {
+            read: true,
+            write: true,
+          },
+          ...config,
+        },
+      },
+      developmentMode,
+    );
+  }
+
+  flexibleSync(developmentMode = true) {
+    return this.mongodbService(
+      {
+        kind: "flexible",
+      },
+      developmentMode,
+    );
+  }
+
+  developmentMode(enabled = true) {
+    return super.sync({ development_mode_enabled: enabled });
+  }
+
+  triggerClientResetFunction() {
+    this.function({
+      name: "triggerClientReset",
+      private: false,
+      run_as_system: true,
+      source: `
+        exports = async function (appId, userId) {
+          return (await deleteClientFile("__realm_sync_" + appId, userId)) || (await deleteClientFile("__realm_sync", userId));
+        };
+        
+        async function deleteClientFile(db, userId) {
+          const mongodb = context.services.get("mongodb");
+          return (await mongodb.db(db).collection("clientfiles").deleteMany({ ownerId: userId })).deletedCount > 0;
+        }
+      `,
+    });
+  }
+}
```

Comment:
I split up the builder pattern in two:
- The app importer package owns an implementation, which is very close to the configuration endpoints exposed by the server.
- This `ExtendedAppConfigBuilder` introduce additional (more opinionated) shorthands for frequently used configurations, such as enabling partition-based or flexible sync.

Reply:
I'd suggest we refactor this when adopting it from the `@realm/react` package. If there are some methods that could be easily reused, I think that makes sense to hoist into the app importer. For now, I just wanted these to be local to the test suite to increase cohesion between the tests and actual app config objects.

Reply:
Makes sense!